Add valid_move? for Bishop

erinmorrissey opened the pull request:
Here's what I have started for the Bishop logic, though between this and the King-specific logic it's not looking terribly DRY. @wdehaes I'm curious to see what you have in mind for the Rook ticket...

lasley commented on app/models/bishop.rb:
```diff
@@ -0,0 +1,10 @@
+class Bishop < Piece
+  def valid_move?(x_destination, y_destination)
+    return false unless super(x_destination, y_destination)
+    return false if obstructed?(x_destination, y_destination)
```
Maybe move obstructed to superclass?

Knight is the only piece that can't be obstructed. You could change the super method signature to:

```
valid_move?(x_destination, y_destination, obstruct=true)
```

Then just `super(x, y, false)` in the Knight & leave everything else as `super(x, y)`. That will help a bit with the DRY.

lasley commented on app/models/bishop.rb:
```diff
@@ -0,0 +1,10 @@
+    return false if obstructed?(x_destination, y_destination)
+    x_diff_val = diff(x_destination, x_position)
+    y_diff_val = diff(y_destination, y_position)
+    return true if x_diff_val == y_diff_val
```
`return x_diff_val == y_diff_val`

erinmorrissey commented:
Comments were very helpful, thanks Dave! I refactored and added tests. Grabbed the Knight ticket so I can incorporate the logic you suggested for `obstructed?`. Also had to adjust `moving_diagonally?` on the Piece class.

wdehaes commented:
LGTM. Merging this PR. @erinmorrissey definitely a good idea to add a call the obstructed? method to the valid_move? method in the piece superclass. Makes it more DRY indeed.
